app/thirdparty/java/other/writeily/format/markdown/WrMarkdownHeaderSpanCreator.java

```diff
@@ -28,36 +26,29 @@ public WrMarkdownHeaderSpanCreator(final CharSequence text, int color, final flo
         _spanCreator = new WrProportionalHeaderSpanCreator(textSize, color);
     }
 
-    public Object callback(Matcher m) {
-        final char[] charSequence = extractMatchingRange(m);
-        float proportion = calculateProportionBasedOnHeaderType(charSequence);
+    public Object callback(final Matcher m) {
+        final float proportion = calculateProportionBasedOnHeaderType(m.start(), m.end());
         return _spanCreator.createHeaderSpan(proportion);
     }
 
-    private char[] extractMatchingRange(Matcher m) {
-        return _text.subSequence(m.start(), m.end()).toString().trim().toCharArray();
-    }
-
-    private Float calculateProportionBasedOnHeaderType(final char[] charSequence) {
-
-        Float proportion = calculateProportionForHashesHeader(charSequence);
+    private float calculateProportionBasedOnHeaderType(final int start, final int end) {
+        final float proportion = calculateProportionForHashesHeader(start);
```

Comment:
No need to make so many copies . We have a reference to the original text anyway